sources/packages/api/src/services/student/student.service.ts

```diff
@@ -324,14 +325,22 @@ export class StudentService extends RecordDataModelService<Student> {
         "student.birthDate",
         "user.firstName",
         "user.lastName",
+        "sinValidation.sin",
       ])
       .leftJoin(
         Application,
         "application",
         "application.student.id = student.id",
       )
       .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
       .where("user.isActive = true");
+
+    if (searchCriteria.sin) {
+      searchQuery.andWhere("sinValidation.sin = :sin", {
```

Comment:
Is there any possibility to this value be received with white spaces?
Maybe it would be good to use the method `removeWhiteSpaces` either way here.

Reply:
Yes, I implemented it. Please view line 341 below.